File: 769-max-chunks-to-make-sorted/769-max-chunks-to-make-sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ans = 0
        maxValue = 0
        
        for i in range(len(arr)):
            maxValue = max(maxValue, arr[i])
            if maxValue == i:
                ans += 1
                
        return ans

File: 769-max-chunks-to-make-sorted/769-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        return solve(arr);
    }
private:
    int solve(vector<int>& arr) {
        int ans = 0;
        int maxValue = 0;
        for(int i = 0; i < arr.size(); ++i) {
            maxValue = max(maxValue, arr[i]);
            if(maxValue == i)
                ++ans;
        }
        return ans;
    }
};